catalog/plugin/catalog-plugin-clavin/src/main/java/org/codice/ddf/catalog/content/plugin/clavin/ClavinPlugin.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.content.plugin.clavin;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.activation.MimeType;
+
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.bericotech.clavin.gazetteer.CountryCode;
+import com.bericotech.clavin.gazetteer.GeoName;
+import com.google.common.net.MediaType;
+
+import ddf.catalog.content.data.ContentItem;
+import ddf.catalog.content.operation.CreateStorageResponse;
+import ddf.catalog.content.plugin.PostCreateStoragePlugin;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.geo.formatter.CompositeGeometry;
+import ddf.geo.formatter.MultiPoint;
+import ddf.geo.formatter.Point;
+
+public class ClavinPlugin implements PostCreateStoragePlugin {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClavinPlugin.class);
+
+    private ClavinWrapper clavinWrapper;
+
+    private Set<MediaType> mediaTypes = new HashSet<>();
+
+    public void setClavinWrapper(ClavinWrapper clavinWrapper) {
+        this.clavinWrapper = clavinWrapper;
+    }
+
+    public ClavinPlugin() {
+        mediaTypes.add(MediaType.MICROSOFT_EXCEL);
+        mediaTypes.add(MediaType.MICROSOFT_WORD);
+        mediaTypes.add(MediaType.MICROSOFT_POWERPOINT);
+    }
```

Comment:
Suggest adding support for the following: `MediaType.OOXML_DOCUMENT`, `MediaType.OOXML_SHEET`, and `MediaType.OOXML_PRESENTATION` to enrich metacards created for `.docx`, `.xlsx`, and `.pptx` documents.